### ISSUE TITLE
API differences?

### ISSUE THREAD

**Issue by kic0** [M5M400/supportxmr-gui#6]
I was testing the new UI and changed the source api in script_min.js from https://supportxmr.com/api/ to https://api.xmrpool.net/ and now doesn't load block info or network and pool Hashrates, are there differences between Snipa "reference" code and sxmr API's that could explain the missing data?

Thanks in advance

**Comment by M5M400** [comment]
are you sure that the requests are not being blocked by your browser or a
plugin like ublock?

**Comment by kic0** [comment]
With SupportXmr api works fine just changed the url in the code to Snipas API and it only blocks part of the site strangely Payments for instance are shown ok

http://node.xmr.pt/

**Comment by M5M400** [comment]
![Screenshot from 2019-09-25 16-46-16](https://user-images.githubusercontent.com/22886679/65612290-4f5b4680-dfb4-11e9-9939-c70b58579c56.png)

We just added two api endpoints (/network/chart/difficulty and /motd). It seems /motd is not yet avail on xmrpool.net - I wonder if it missing is breaking the js, but it should be unrelated.

@Thunderosa ?

**Comment by Thunderosa** [comment]
Well, so there are some very specific data things happening with this UI that would be pretty sensitive to any API differences.

The motd error is self explainatory,...but eff and tme missing is something else. Without being able to see exactly where it's choking, I'm guessing something with the blocks endpoint specificially. I think you were on the Pool Blocks page when you got this error?

If so, for the first page of results the blocks in the list and the blocks in the chart are the same data source and then the blocks list starts using it's own lookup on page 2.

So that would mean your issue is probably in the 'block' mode of the data function (as opposed to 'blockhistory').

That section requires the following endpoints from the API:
'hash', 'height', 'value' (in atomic units), 'shares', 'diff' & 'valid'

$D['block'] being populated would be the object to check first.

**Comment by Thunderosa** [comment]
Nope, nevermind,...you found a good bug. It's because you haven't found a block within the window of time. I'll correct this for the next version.

For the motd error a quick fix would be to comment out the line: url = 'pool/motd'; If that url variable isn't set then it won't try to fix. If you decide you want the feature, you can setup the endpoint and restore the value.

**Comment by kic0** [comment]
Awesome thanks for the help! I'll update and let you know :)
